### ISSUE TITLE
repeated errors with simple snap project

### ISSUE THREAD

**Issue by asheshambasta** [rikvdkleij/intellij-haskell#254]
Seeing hundreds of issues like this in the intellij event log:

```
18:13	[project-stack-repl-snapbare:exe:snapbare] Stack repl could not be started within 5 seconds

18:13	[project-stack-repl-snapbare:exe:snapbare] Executable named intero not found on path: ["/Users/asheshambasta/code/haskell/snapbare/.stack-work/install/x86_64-osx/lts-10.3/8.2.2/bin","/Users/asheshambasta/.stack/snapshots/x86_64-osx/lts-10.3/8.2.2/bin","/Users/asheshambasta/.stack/compiler-tools/x86_64-osx/ghc-8.2.2/bin","/Users/asheshambasta/.stack/programs/x86_64-osx/ghc-8.2.2/bin","/Users/asheshambasta/.local/bin","/usr/local/opt/jpeg-turbo/bin","/usr/bin","/bin","/usr/sbin","/sbin","/usr/local/bin","/Users/asheshambasta/.rvm/bin"]

18:13	[project-stack-repl-snapbare:exe:snapbare] Executable named intero not found on path: ["/Users/asheshambasta/code/haskell/snapbare/.stack-work/install/x86_64-osx/lts-10.3/8.2.2/bin","/Users/asheshambasta/.stack/snapshots/x86_64-osx/lts-10.3/8.2.2/bin","/Users/asheshambasta/.stack/compiler-tools/x86_64-osx/ghc-8.2.2/bin","/Users/asheshambasta/.stack/programs/x86_64-osx/ghc-8.2.2/bin","/Users/asheshambasta/.local/bin","/usr/local/opt/jpeg-turbo/bin","/usr/bin","/bin","/usr/sbin","/sbin","/usr/local/bin","/Users/asheshambasta/.rvm/bin"]
```

![image](https://user-images.githubusercontent.com/580365/35196858-15b8c664-fed8-11e7-932e-b12f8bfd18ef.png)

The project is a barebones project init with `snap init barebones` and then changed to stack [as described here](https://stackoverflow.com/a/36588040/1360368).

The logs don't seem to stop even without anything and the task "building project" keeps getting restarted over and over again.

Unfortunately, the only way I've found to stop this is by disabling the plugin.

Interacting with the project with the command line seems to be ok:
`stack ghci` etc. all seem to work.

**Comment by rikvdkleij** [comment]
For some reason `intero` could not be build. Try to find in the Event log where `intero` is build to see the problem.

**Comment by Sir4ur0n** [comment]
Is the problem still happening? If yes, please provide steps to reproduce.

**Comment by declension** [comment]
FYI I get this exact symptom with new Stackage `lts` versions, and yes because intero isn't compatible with these packages (see https://github.com/commercialhaskell/intero/issues/586)

**Comment by rikvdkleij** [comment]
@declension Maybe this helps https://github.com/commercialhaskell/intero/issues/584

**Comment by declension** [comment]
> @declension Maybe this helps [commercialhaskell/intero#584](https://github.com/commercialhaskell/intero/issues/584)

Thanks. 
FYI I filed a bug at https://github.com/commercialhaskell/intero/issues/586 - and [it's now fixed for GHC 8.4.4](https://github.com/commercialhaskell/intero/pull/587) (whilst people work on that 8.6.1 support)

**Comment by rikvdkleij** [comment]
@asheshambasta Thanks for you reply but that is not the case for the intellij plugin. intero gets build for each project. It is not installed globally.

**Comment by asheshambasta** [comment]
@rikvdkleij Okay, is that a recent change? I vaguely remember having to input the paths for the intero binary (along with some others) when configuring the plugin. I'm not sure about that though.

**Comment by rikvdkleij** [comment]
@asheshambasta No, never been the case. Only for Haskell tools like Hlint one had to configure the paths. But for we recent betas that is also not the case anymore.

**Comment by rikvdkleij** [comment]
intero has to be build with same GHC/Stackage version as project.

**Comment by asheshambasta** [comment]
@rikvdkleij ah okay, thanks for clarifying that. My experience in this regard has been patchy. Apologies for that.

**Comment by rikvdkleij** [comment]
@asheshambasta No problem.

I have to say that last of couple of weeks I'm running into various stack issues which only seems to be solved by deleting `~/.stack` and `.stack-work` folders. Or by switching between stack, nix and docker and hope that something works....

But currently I have an issue in which even that does not solve the problem....

**Comment by asheshambasta** [comment]
@rikvdkleij Does something like `stack init --reinstall` help? I had a similar problem and doing that seemingly fixed the problem for me.

**Comment by rikvdkleij** [comment]
@asheshambasta Thanks! No, that is not the problem. It's the problem that stack binary can not download files from `s3.amazonaws.com`.

**Comment by asheshambasta** [comment]
@rikvdkleij if you can post some of the error messages here maybe I can try to repro the issue and help.

**Comment by cdfa** [comment]
I'm having the same issue! I just freshly installed everything. What should i do to fix this? I'm using lts-12,18 as resolver (which uses ghc 8.4.4 according to [this](https://www.stackage.org/lts-12.18) (sorry, i'm quite new to all the tools around haskell).

**Comment by rikvdkleij** [comment]
@cdfa Check that you can build intero on your system. For example on Ubuntu you need to install `libtinfo-dev` with `sudo apt-get install libtinfo-dev`

**Comment by cdfa** [comment]
I don't think that's the problem. This the output i get from `stack install intero`:
```
Populated index cache.
Didn't see ghci-8.4.4 in your package indices.
Updating and trying again.
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/
Downloading timestamp
No updates to your package index were found
Update complete
The following package identifiers were not found in your indices: ghci-8.4.4
Possible candidates: ghci-8.4.3, ghci-8.4.1, ghci-8.4.2.
```

**Comment by cdfa** [comment]
I get quite similar output:
```
Didn't see ghci-8.4.4 in your package indices.
Updating and trying again.
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/
Downloading timestamp
Downloading snapshot
Updating index
Updated package index downloaded
Update complete
Populated index cache.
The following package identifiers were not found in your indices: ghci-8.4.4
Possible candidates: ghci-8.4.3, ghci-8.4.1, ghci-8.4.2.
```

**Comment by cdfa** [comment]
It seems [ghc-8.4.4](http://hackage.haskell.org/package/ghci) is indeed not on hackage, but it is installed in `~/stack/programs/x86_64-linux/`. Why is it not just using that?

**Comment by rikvdkleij** [comment]
`intero` is build with same Stackage version as defined in your project's `stack.yaml`

**Comment by cdfa** [comment]
Which is `lts-12.18`, which indeed targets ghc-8.4.4, so that seems correct

**Comment by rikvdkleij** [comment]
Sometimes it helps to delete your `~/.stack` and  the`.stack-work` in your project's folder.

**Comment by cdfa** [comment]
Tried it, it reinstalled the ghc, but did't work

**Comment by rikvdkleij** [comment]
You can try to get help on the gitter channel. I saw that @develop7 was involved in this intero issue.

**Comment by develop7** [comment]
Could you show the `stack.yaml` and `.cabal` files? The SO link above is dead

**Comment by cdfa** [comment]
I generated the project with stack new, so it should be pretty standard. I think I added some extra deps to make it compile at some point, but I'm not sure.
[stack.yaml](https://pastebin.com/HZgD0J0B)
[.cabal](https://pastebin.com/KtSVU2yz) 
I did manage to get it working in the end by creating a new project with `stack new` again and this time it worked. Maybe the stack version that's distributed with the haskell platform is not compatible?
